Build r-seurat 4.4.0 for r-base 4.4

jdblischak opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

---

Closes #47 

cc: @mojaveazure, @johnkerl

The goal of this PR is to create a linux-64 conda binary for the legacy version 4.4.0 of Seurat against the more recent R 4.4.

I purposefully didn't bump the build number. Since Seurat 4.4.0 has never been built for R 4.4 previously, only this single binary will be uploaded with the build string `r44********_0`. Bumping the build number to `1` would imply that a build `0` exists, which I think would be confusing.

I created the internal branch `4.4.0` to support these backporting efforts from the commit https://github.com/conda-forge/r-seurat-feedstock/commit/8b235b3d99b1738915c267efcb59985802455e4a (no commits were added between [4.4.0 and 5.0.0](https://github.com/conda-forge/r-seurat-feedstock/compare/8b235b3d99b1738915c267efcb59985802455e4a...832b5cb9a63709b11b28cb0ea213e4792a06a324)). I created an empty commit with `[ci skip]` because the Azure builds immediately fail since the CI configuration is so outdated.

```sh
git checkout -b 4.4.0 8b235b3d99b1738915c267efcb59985802455e4a
## Switched to a new branch '4.4.0'
git log -n 1 --pretty=reference
## 8b235b3 ([bot-automerge] r-seurat v4.4.0 (#35), 2023-09-28)
git commit --allow-empty -m "[ci skip] Start backport branch for 4.4.0"
## [4.4.0 a2a7556] [ci skip] Start backport branch for 4.4.0
git push upstream 4.4.0
```

I updated the requirements to match the [4.4.0 `DESCRIPTION` file](https://github.com/satijalab/seurat/blob/v4.4.0/DESCRIPTION):

```sh
conda skeleton cran --version 4.4.0 r-seurat
```

jdblischak committed:
Limit to single variant: r-base 4.4 on linux-64

jdblischak committed:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.46.1, and conda-forge-pinning 2025.03.05.16.19.52

jdblischak committed:
Remove requirements for win-64 and cross-compiled builds

jdblischak committed:
Update requirements to match seurat 4.4.0 DESCRIPTION

https://github.com/satijalab/seurat/blob/v4.4.0/DESCRIPTION

conda skeleton cran --version 4.4.0 r-seurat

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `{{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler. For further details, please see https://github.com/conda-forge/conda-forge.github.io/issues/2102.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13684361525](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13684361525). Examine the logs at this URL for more detail.</sub>

jdblischak committed:
Fix lint: add {{ stdlib("c") }}

jdblischak committed:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.46.1, and conda-forge-pinning 2025.03.05.16.19.52

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

mfansler reviewed (commented):
Wondering if we need to be concerned about any of this?

```r
> library('Seurat')
Attaching SeuratObject
‘SeuratObject’ was built under R 4.4.1 but the current version is
4.4.3; it is recomended that you reinstall ‘SeuratObject’ as the ABI
for R may have changed
‘SeuratObject’ was built with package ‘Matrix’ 1.6.5 but the current
version is 1.7.2; it is recomended that you reinstall ‘SeuratObject’ as
the ABI for ‘Matrix’ may have changed
Seurat v4 was just loaded with SeuratObject v5; disabling v5 assays and
validation routines, and ensuring assays work in strict v3/v4
compatibility mode
```

We've definitely had [Matrix ABI issues](https://github.com/conda-forge/r-matrix-feedstock/issues/42) in other feedstocks before, and IIRC they only crop up in function calls. You probably should test the artifact or add some unit tests to the recipe.

Could turn out we need at least a new `r-seuratobject` build as well.

jdblischak committed:
Require r-seuratobject >=5.0.2

r-seurat built with r-seuratobject 4 can be installed at runtime
with r-seuratobject 4 or 5.

r-seurat build with r-seuratobject 5 can only be installed at runtime
with r-seuratobject 5.

One option would be to force building with r-seuratobject 4, but this
older version is not available for R 4.4. Thus just updating to the
newer r-seuratobject 5.0.2

mojaveazure commented:
Hi @mfansler, former lead maintainer of Seurat/SeuratObject here. The R 4.4.1 -> 4.4.3 is nothing to be concerned about, and the Seurat v4 + SeuratObject v5 message is just informative saying that additional v5 functionality is being disabled and SeuratObject is running in "v4 mode" (not important to anyone involved here)

However the Matrix 1.6.5 -> 1.7.2 is something a bit more concerning. I haven't looked too closely into the 1.7 release of Matrix, but rebuilding `r-seuratobject` against Matrix 1.7.x would be the safe option

jdblischak commented:
Waiting on rebuilding r-seuratobject with the latest r-base and r-matrix https://github.com/conda-forge/r-seuratobject-feedstock/pull/21

mfansler reviewed (approved):
LGTM!

FYI, load test now shows:

```r
> library('Seurat')
Attaching SeuratObject
Seurat v4 was just loaded with SeuratObject v5; disabling v5 assays and
validation routines, and ensuring assays work in strict v3/v4
compatibility mode
```

jdblischak commented:
Thanks for the review and merge @mfansler!

Documenting the exact version used in the [Azure build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1191740&view=logs&j=7b6f2c87-f3a7-5133-8d84-7c03a75d9dfc&t=9eb77fd2-8ddd-5444-8fc0-71cb28dcb736) from the merge commit https://github.com/conda-forge/r-seurat-feedstock/commit/cff5d1815f662cf382c650e8a401f10c0aef2cb9:

* r-base: 4.4.3-h9c32bc2_0
* r-matrix: 1.7_2-r44h2ae2be5_0
* r-seuratobject: 5.0.2-r44h93ab643_2 (this is build number 2 from https://github.com/conda-forge/r-seuratobject-feedstock/pull/21)
